vllm/model_executor/layers/fused_moe/layer.py

```diff
@@ -203,8 +326,16 @@ def forward_hpu(
             topk_weights = topk_weights.to(x.dtype)
 
         final_hidden_states = torch.zeros_like(x)
-        num_experts = layer.w13_weight.shape[0]
-        n_expert_slice = layer.w13_weight.shape[0] // 8
+        num_experts = layer.num_experts
+        if hasattr(layer, "w13_weight") and layer.w13_weight is not None:
+            assert (
+                layer.w13_weight.shape[0] == num_experts
+            ), f"Expected {layer.w13_weight.shape[0]} experts, got {num_experts}"
+        # For mixtral, the `num_expert_group` is 8.
+        if num_expert_group is None:
+            num_expert_group = 8
```

Comment:
Can we load this from the model config instead of hardcoding?

Reply:
These changes are not used in this PR, so I have reverted them.